var/spack/repos/builtin/packages/py-pennylane-lightning/package.py

```diff
@@ -7,20 +7,128 @@
 from spack.package import *
 
 
-class PyPennylaneLightning(PythonPackage):
-    """PennyLane-Lightning plugin"""
+class PyPennylaneLightning(CMakePackage, PythonExtension):
```

Comment:
Since the package is now a CMake tool with optional Python bindings, I think it would make sense to rename it to `pennylane-lightning`. Another option would be to have both `pennylane-lightning` and `py-pennylane-lightning` packages where the former is the CMake part and the latter is only the Python bindings and depends on the former. Not sure if it's possible to build your Python bindings with a separately installed CMake part or if they have to be installed simultaneously.

Reply:
It is a good question. Right now, we have an explicitly tight coupling with the Python package, and so that is our delivery and consumption mechanism. We expect this to remain the case for most of 2023, so likely separating this out will require some decent effort on our end.

I would be keen to keep the structure as is for now, assuming this isn't a problem.

Reply:
I don't think `py-pennylane-lightning~python` makes sense. See https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#choosing-a-package-name for the official docs. I think it makes the most sense to rename the package. We can think about splitting them into separate packages later.

Reply:
Since the package comes from https://pypi.org/project/PennyLane-Lightning/ as the intended use, the reason to disable the Python layer is simply to speed up compilation to generate test and benchmarking binaries at the CPP layer. It is not possible to use the compiled library itself without the Python layer, so it isn't optional in this case.

The internal kernels can be compiled separately and validated, hence the `+python` flag option in this instance.
I think keeping the Python naming makes sense as it seems to fall into the same structure as listed on the link with py-genders  py-py2cairo, py-pygobject, etc.

Reply:
I'll discuss with our team internally and see if this split makes sense now rather than later.

Reply:
In that case I'm confused what the point of this PR is. You're adding a great deal of complexity to the package but the end goal is still to build the exact same Python library. I was under the impression that the change to the CMakePackage base class was to support greater flexibility and allow for C++-only installations, but it seems like the thing getting installed is still the same. We could remove the `+python` variant and always installs the Python portion I suppose.

Reply:
Having access to Spack in this case allows us to build the Python module with the configurability introduced by Spack.

We have tested manually compiling the package on a variety of HPC systems through different compilers, and having the variants allows us to tailor-suit the Python package for each respective system (ie without or without AVX512 support, OpenMP, Kokkos, etc).

When choosing a set of variants, the binary options give us the up-front information to understanding if the full test-suite passes on the CPP side, as well as how the kernels perform under the hood.

I'm fine dropping the +python variant and always ensuring the Python module is built, if you think this is the best situation here. We do not plan to separate out the C++ only components into a separate module yet, as this also requires plenty of retooling on the package itself. It may be a future change, but not part of our roadmap just yet. Hope that clarifies.

Reply:
Sounds good, let's just drop `+python` for now.

Reply:
Sure thing, and thanks for the discussion on this.